examples/reminders/lib/base/data_sources/local/objectbox_local_data_source.dart

```diff
@@ -0,0 +1,179 @@
+
+  static const _authorId = 'authorId';
+  static const _anonymous = 'anonymous';
+
+  static ObjectBoxLocal? _instance;
+
+  ObjectBoxLocal._init(this._storage, this._remindersBox);
+
+  static ObjectBoxLocal getInstance(
+    FlutterSecureStorage storage,
+    EnvironmentConfig env,
+  ) {
+    if (_instance == null) {
+      final store = openStore();
+      final remindersBox = _initCloud(env, store);
+      _instance = ObjectBoxLocal._init(storage, remindersBox);
+    }
+    return _instance!;
+  }
+
+  static Future<Box<ObjectBoxLocalReminderModel>> _initCloud(
+      EnvironmentConfig config, Future<Store> openStore) async {
+    var store = await openStore;
+    var reminderBox = store.box<ObjectBoxLocalReminderModel>();
+
+    return reminderBox;
+  }
+
+  @override
+  Future<ReminderModel> create(
+      {required String title,
+      required DateTime dueDate,
+      required bool complete}) async {
+    var authorId = await _getAuthorIdOrNull();
+
+    final reminder = ObjectBoxLocalReminderModel(
+      title: title,
+      dueDate: dueDate,
+      complete: complete,
+      authorId: authorId,
+    );
+
+    (await _remindersBox).put(reminder);
+    return reminder;
+  }
+
+  @override
+  Future<void> delete(String id) async {
+    (await _remindersBox).remove(int.parse(id));
+  }
+
+  ///Return all reminders from the ObjectBox database
+  @override
+  Future<ReminderListResponse> getAll(ReminderModelRequest? request) async {
+    final authorId = await _getAuthorIdOrNull();
+    final result = (await _remindersBox)
+        .query(getCondition(request, authorId))
+        .order(ObjectBoxLocalReminderModel_.dueDate,
+            flags: getOrder(request?.sort))
+        .build()
+        .find();
+    return ReminderListResponse(items: result, totalCount: result.length);
+  }
+
+  ///Return only the incomplete reminders that are before the current date
+  @override
+  Future<ReminderListResponse> getAllDashboard(
+      ReminderModelRequest? request) async {
+    final authorId = await _getAuthorIdOrNull();
+    final result = (await _remindersBox)
+        .query(getCondition(request, authorId))
+        .order(ObjectBoxLocalReminderModel_.authorId,
+            flags: getOrder(request?.sort))
+        .build()
+        .find();
+    return ReminderListResponse(items: result);
+  }
+
+  @override
+  Future<int> getCompleteCount() async => (await _remindersBox)
+      .query(ObjectBoxLocalReminderModel_.complete.equals(true))
+      .build()
+      .find()
+      .length;
+
+  @override
+  Future<int> getIncompleteCount() async => (await _remindersBox)
+      .query(ObjectBoxLocalReminderModel_.complete.equals(false))
+      .build()
+      .find()
+      .length;
+
+  @override
+  Future<ReminderPair> update(ReminderModel updatedModel) async {
+    var oldModel = (await _remindersBox).get(int.parse(updatedModel.id));
+    final newModel = (await _remindersBox).put(
+      ObjectBoxLocalReminderModel(
```

Comment:
From what I see here we have issues with the naming convention and the approach in general.

I believe the `ObjectBoxLocalReminderModel` is not an actual business model, but rather a data transfer object (DTO) used for the object box communication. The same as we use a HashMap in the API communication.

If this is true, please implement the industry established software design pattern described here

DTO (ObjectBox)  -> Mapper -> Business Model
Business Model    -> Mapper -> DTO (ObjectBox)


https://www.baeldung.com/java-dto-pattern

Reply:
Done

Reply:
Done